inc/targetbase.class.php

```diff
@@ -388,6 +390,55 @@ protected function setTargetCategory($data, $formanswer) {
          case self::CATEGORY_RULE_SPECIFIC:
             $category = $this->fields['category_question'];
             break;
+         case self::CATEGORY_RULE_LAST_ANSWER:
+            $category = null;
+            $form_id = $formanswer->fields['id'];
+
+            // Get all answers for dropdown questions of this form, ordered
+            // from last to first displayed
+            $answers = $DB->request([
+               'SELECT' => ['answer.answer', 'question.values'],
+               'FROM' => PluginFormcreatorAnswer::getTable() . ' AS answer',
+               'JOIN' => [
+                  PluginFormcreatorQuestion::getTable() . ' AS question' => [
+                     'ON' => [
+                        'answer' => 'plugin_formcreator_questions_id',
+                        'question' => 'id',
+                     ]
+                  ]
+               ],
+               'WHERE' => [
+                  'answer.plugin_formcreator_formanswers_id' => $form_id,
+                  'question.fieldtype'                       => "dropdown",
+               ],
+               'ORDER' => [
+                  'row DESC',
+                  'col DESC',
+               ]
+            ]);
+
+            foreach ($answers as $answer) {
+               // Decode dropdown settings
+               $values = json_decode($answer['values']);
```

Comment:
in some cases the value may not be a JSON. I recently added a method to get the itemtype : PluginFormcreatorDropdownField::getSubItemtype()

To instanciate the field:

```php
$question = new PluginFormcreatorQuestion();
$question->getFromDB($questinId); // need to retrieve the id from the request a few lines earlier
$field = PluginFormcreatorFields::getFieldInstance($question->fields['fieldtype'], $question);
$itemtype = $field->getSubItemtype();
```

If you have an eligible question populated by the user then hidden by a condition (when the user fills other questions), I'm not sure that the value will be empty in DB then ignored by this code.

Reply:
In this case I find it a bit non optimal to load two news objects from DB in order to access a value that I already have.
I've made a new static method in 5240c38 to avoid this issue, let me know if that's not ok for you.

Reply:
Ok for this commit.